tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -295,10 +282,158 @@ def finalizer():
-        logger.info('RWX Test')
+    def test_rwx_dynamic_pvc(self, setup_base):
+        """
+        RWX Dynamic PVC creation tests with Reclaim policy set to Delete/Retain
+        """
+        logger.info(f"CephFS RWX test")
+        logger.info(f"Creating two pods using same PVC {self.pvc_obj.name}")
+        logger.info(
+            f"Creating first pod on node: {self.worker_nodes_list[0]} "
+            f"with pvc {self.pvc_obj.name}"
+        )
+        pod_obj1 = helpers.create_pod(
+            interface_type=self.interface_type, pvc_name=self.pvc_obj.name,
+            desired_status=constants.STATUS_RUNNING, wait=True,
+            namespace=self.namespace, node_name=self.worker_nodes_list[0],
+            pod_dict_path=constants.NGINX_POD_YAML
+        )
+        node_pod1 = pod_obj1.get().get('spec').get('nodeName')
+
+        logger.info(
+            f"Creating second pod on node: {self.worker_nodes_list[1]} "
+            f"with pvc {self.pvc_obj.name}"
+        )
+
+        pod_obj2 = helpers.create_pod(
+            interface_type=self.interface_type, pvc_name=self.pvc_obj.name,
+            desired_status=constants.STATUS_RUNNING, wait=True,
+            namespace=self.namespace, node_name=self.worker_nodes_list[1],
+            pod_dict_path=constants.NGINX_POD_YAML
+        )
+        node_pod2 = pod_obj2.get().get('spec').get('nodeName')
+
+        assert node_pod1 != node_pod2, 'Both pods are on the same node'
+
+        # Run IO on both the pods
+        logger.info(f"Running IO on pod {pod_obj1.name}")
+        file_name1 = pod_obj1.name
+        logger.info(file_name1)
+        pod_obj1.run_io(
+            storage_type=self.storage_type, size=self.io_size, runtime=30,
+            fio_filename=file_name1
+        )
+
+        logger.info(f"Running IO on pod {pod_obj2.name}")
+        file_name2 = pod_obj2.name
+        pod_obj2.run_io(
+            storage_type=self.storage_type, size=self.io_size, runtime=30,
+            fio_filename=file_name2
+        )
+
+        # Check IO and calculate md5sum of files
+        pod.get_fio_rw_iops(pod_obj1)
+        md5sum_pod1_data = pod.cal_md5sum(
+            pod_obj=pod_obj1, file_name=file_name1
+        )
+
+        pod.get_fio_rw_iops(pod_obj2)
+        md5sum_pod2_data = pod.cal_md5sum(
+            pod_obj=pod_obj2, file_name=file_name2
+        )
+
+        logger.info(f"verify data from alternate pods")
+
+        assert pod.verify_data_integrity(
+            pod_obj=pod_obj2, file_name=file_name1,
+            original_md5sum=md5sum_pod1_data
+        )
+
+        assert pod.verify_data_integrity(
+            pod_obj=pod_obj1, file_name=file_name2,
+            original_md5sum=md5sum_pod2_data
+        )
+
+        # Verify that data is mutable from any pod
+
+        logger.info(f"Perform modification of files from alternate pod")
+        # Access and rename file written by pod-2 from pod-1
+        file_path2 = pod.get_file_path(pod_obj2, file_name2)
```

Comment:
For better ordering, please check pod1 before pod2

Reply:
@elad this can be done. But here, the existence of file is validated first  from pod1 itself as you suggested. Its just that the file being validated is file_name2 - written actually by pod2

Reply:
Thanks, now I see